### ISSUE TITLE
Loading CMIP6 data

### ISSUE THREAD

**Issue by ledm** [ESMValGroup/ESMValCore#34]
Hi everone,

Related to the PR ESMValGroup/ESMValTool#1015. That PR address the problems with UKESM & MIROC6. The problem both of these models was that they used the terms `latitude` and `longitude` instead of `lat` & `lon`. 

The other models that are available now on JASMIN are:
- CESM2
- CNRM-CM6-1
- GISS-E2-1-G

These three all seem to work fine with ESMValTool at the moment!

The models below don't work at the moment, and I'm not sure how to address their problems.
- BCC-CSM2-MR
- BCC-ESM1
- IPSL-CM6A-LR

The two BCC models use irregular grids, however they haven't used the standard terms to describe latitude, longitude, cell index along first dimension and cell index along second dimension (`lat`, `lon` and `i` ,`j`). Instead they have used: `lat`, `lon` and `latitude` ,`longitude`). Basically, this is rather confusing because `latitude` is a 2D array and `lat` is a one D array. I don't think iris is very happy with this dataset. This causes the following error message:
```
esmvaltool.cmor.check.CMORCheckError: There were errors in variable tos:
tos: does not match coordinate rank: 3 != 1
```

The IPSL dataset includes the terms `nav_lat` and `nav_lon` instead of `lat` and `lon`, which is annoying but not the end of the world. The files I'm looking at also include an `area` array (instead of `areacello`) which causes trouble when loading the cube, because it wants to load the netcdf as a list of cubes instead of as a single cube. This produces the error:
```
ValueError: Cubes were not reduced to one afterfixing: 
0: cell_area / (m2)                    (-- : 332; -- : 362)
1: sea_surface_temperature / (degC)    (time: 1980; -- : 332; -- : 362)
```

Anyone have any ideas how to address these issues?

**Comment by npgillett** [comment]
@ledm Is your fix for the BCC models now included in the standard release of esmvalcore? I installed it from conda and updated last week, but when I try to read in tos from BCC-CSM2-MR, I get:
```
esmvalcore.cmor.check.CMORCheckError: There were errors in variable tos:
tos: does not match coordinate rank
in cube:
sea_surface_temperature / (degC)    (time: 1980; latitude: 232; longitude: 360)
     Dimension coordinates:
          time                           x               -               -
          latitude                       -               x               -
          longitude                      -               -               x
     Auxiliary coordinates:
          latitude                       -               x               x
          longitude                      -               x               x

```
Looking at the thread above, it looks like you fixed this on 15th April last year. Thanks for any help anyone can give on this. My log file is attached for reference.
[main_log_debug.txt](https://github.com/ESMValGroup/ESMValCore/files/4314824/main_log_debug.txt)

**Comment by schlunma** [comment]
@npgillett a fix for this will be available soon, see #560

**Comment by bouweandela** [comment]
This is a very old issue that deals with many different topics. @ledm If any of this is still relevant, can you please open a new issue? Preferably using the new template for data issues?
